docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx

```diff
@@ -248,40 +248,47 @@ First, review the `import` and `export` statements:
 - The `export` statement creates a placeholder empty component.
 
 ```ts ignore
-  1 import { useEffect, useState } from 'react';
-  2 import './reactCOIServiceWorker';
-  3 import ZkappWorkerClient from './zkappWorkerClient';
-  4 import { PublicKey, Field } from 'o1js';
-  5 import GradientBG from '../components/GradientBG.js';
-  6 import styles from '../styles/Home.module.css';
-  7 
-  8 let transactionFee = 0.1;
-  9 
- 10 export default function Home() {
- 11   return <div/>
- 12 }
+import { useEffect, useState } from 'react';
+import './reactCOIServiceWorker';
+import ZkappWorkerClient from './zkappWorkerClient';
+import { PublicKey, Field } from 'o1js';
+import GradientBG from '../components/GradientBG.js';
+import styles from '../styles/Home.module.css';
+
+let transactionFee = 0.1;
+const ZKAPP_ADDRESS = 'B62qo2Be4Udo5EG1ux9yMJVkXe9Gz945cocN7Bn4W9DSYyeHZr1C3Ea';
+
+export default function Home() {
+  return <div/>
+}
 ```
 
+:::info
+
+For best results, deploy the zkApp to Berkeley yourself, as this test network is sometimes being reloaded. When deployed, replace `ZKAPP_ADDRESS` with the address of your deployment.
+
+:::
+
 ### Add state
 
 This `export` statement creates mutable state that you can reference in the UI. The state updates as the application runs:
 
 ```ts ignore
 ...
- 10 export default function Home() {
- 11   let [state, setState] = useState({
- 12     zkappWorkerClient: null as null | ZkappWorkerClient,
- 13     hasWallet: null as null | boolean,
- 14     hasBeenSetup: false,
- 15     accountExists: false,
- 16     currentNum: null as null | Field,
- 17     publicKey: null as null | PublicKey,
- 18     zkappPublicKey: null as null | PublicKey,
- 19     creatingTransaction: false,
- 20   });
- 21 
- 22   const [displayText, setDisplayText] = useState('');
- 23   const [transactionlink, setTransactionLink] = useState('');
+export default function Home() {
```

Comment:
thanks for these updates @qwadratic 
I defer to  @ymekuria for the code update review and to verify that the tutorial runner does not have line number dependencies

Reply:
@ymekuria  can you also evaluate https://github.com/o1-labs/docs2/issues/694